crypto/src/main/java/net/consensys/cava/crypto/sodium/Signature.java

```diff
@@ -211,6 +211,21 @@ public Bytes sign(Bytes content) {
       return Signature.signDetached(content, this);
     }
 
+    /**
+     * converts signature key (Ed25519) to a box key (Curve25519) so that the same key pair can be used both for
+     * authenticated encryption and for signatures. See https://libsodium.gitbook.io/doc/advanced/ed25519-curve25519
+     *
+     * @return Box.KeyPair generated from Signature.KeyPair.SecretKey
+     */
+    public Box.KeyPair toBoxKeyPair() {
```

Comment:
As a matter of style, I'd prefer this as `Box.KeyPair.forSignatureKey(Signature.SecretKey key)`

Reply:
I thought about adding that as well.  Should I add there and remove from Signature.KeyPair?

Reply:
That would be my preference. E.g. 297f00e1f03586058e4267c0fbbfcc70fe81fcb1

Reply:
Correction - d20454975f15d27d7dcf555dd8210b34cb2597ff

Reply:
Thanks for the suggestion, this is much better.  Moved this to Box.KeyPair and renamed method.